### DISCUSSION TITLE
USG-T Recharge Concentration Package

### DISCUSSION THREAD

**Discussion by roh-sgm** [modflowpy/flopy#1990] [Q&A] [answered]
Hi all, 

Quick question. Was wondering if Flopy is able to load a USG-T Flow and Transport Model?.

The .load instruction dies when trying to read in the RCH package. Pressumably because it's has transport information in it.

![image](https://github.com/modflowpy/flopy/assets/131321724/308bf844-fea4-4e8d-b2a9-d5eacce53f2e)

![image](https://github.com/modflowpy/flopy/assets/131321724/e7b23d82-740d-4aef-a496-c3d0c393df65)

![image](https://github.com/modflowpy/flopy/assets/131321724/c7ebd031-d145-4bb0-8434-79653219694a)

I tested this using Flopy 3.3.6 (the images pasted here), but also I just tested with Flopy 3.4.3 and got the same issue.

PS: first time posting a question. Apologies if I missed any format requirements in this post.

Cheers!
Rodrigo.

**Comment by javgs-bd** [comment]
Hi Rodrigo
Try it with the load-only argument. It might let you load it without the rich package.
As far as I can remember, there was a closed issue in the flopy repo that discussed this. I would point you to it but am not at the computer at the moment.

**Reply by javgs-bd** [reply]
#1300 

That one!

**Reply by roh-sgm** [reply]
Thanks for your suggestion Javier and the link to check the issue #1300.

With the "load_only=[]" argument I was able to skip the RCH package and load DISU, then I modified the list and added other basic packages like: BAS6, CHD, DRN, etc. However, It looks like CLN-TIB/BCT/PCB are also not yet supported by Flopy.

Another issue I found happened when I tried to plot my grid using "dis.plot()". Instead of the plot I got an error:
![image](https://github.com/modflowpy/flopy/assets/131321724/f150edb8-df7c-40f9-bd15-6fb425e757e2)

Have you run into issues like this as well?

Cheers.

**Comment by cnicol-gwlogic** [comment, marked as answer]
Hi,

See issue #1476 for discussion on usg-transport support - no it is not (yet) supported in flopy for reasons provided in that issue. I have started to make progress on setting up a usg-transport repo with tagged releases etc to allow flopy support, but I've really been too busy to commit much time to it, so it's not ready yet. 

On your dis.plot() issue - 1. should it not be disu.plot()? And 2. Have you built a modelgrid object for the model you loaded? There is a function in flopy to do this from a gsf file if you have one to hand.

**Reply by christianlangevin** [reply]
[This flopy example](https://flopy.readthedocs.io/en/latest/Notebooks/mfusg_freyberg_example.html) shows how to build a modelgrid from a gsf file for subsequent plotting.

**Comment by roh-sgm** [comment]
Thanks @cnicol-gwlogic and @langevin-usgs for your replies.
I am closing the issue as I got the answers I was looking for.
Cheers.
